lab-bundle-spec.md

```diff
@@ -308,20 +308,140 @@ environment:
 
-Lab bundles will provisionally support the JSON representation of Activity Tracking currently used in the Qwiklabs web interface. The JSON definition should be stored in file separately from (and referenced directly in) `qwiklabs.yaml`.
+A lab has an Assessment, which in turn contains the Steps (checkpoints).
 
-```yaml
-entity_type: Lab
+#### Assessment
 
-...
+attribute          | required | type    | notes
+-------------------| -------- | --------| --------------------------------------
+passing_percentage | ✓        | integer | The percentage of total points the student must achieve to "pass" the lab.
+steps              | ✓        | array   | An array of [Steps](#steps)
 
-activity_tracking: ./assessment/activity_tracking.json
+```yml
+assessment:
+  passing_percentage: 75
+  steps: ...
 ```
 
-> **Note:** Support for this format should be considered deprecated.
->
-> Further work will be done to define a new DSL for expressing Activity Tracking logic. Exploratory sketches of this DSL can be found in the `./examples` directory of this repository.
+#### Steps
+
+attribute        | required | type                       | notes
+-----------------| -------- | ---------------------------| --------------------------------------
+title            | ✓        | locale dictionary          |
+maximum_score    | ✓        | integer                    | The maximum number of points this step can award.
+student_messages | ✓        | array of objects           | The keys are how the messages will be referenced in the code, and the values are locale dictionaries.
+services         | ✓        | array of resource services | An array of services that will be used in the code block. Each resource type specifies a set of allowed services.
+code             | ✓        | string                     | Code to be executed. See [below](#code) for more information.
+
+```yml
+assessment:
+  passing_percentage: 75
+  steps:
+    - title:
+        locales:
+          en: Create a Cloud Storage bucket
+          es: Crear un depósito de almacenamiento en la nube
+      maximum_score: 5
+      student_messages:
+        - success:
+            locales:
+              en:
+                Great job! You created the bucket!
+              es:
+                ¡Gran trabajo! ¡Creaste el cubo!
+        - bucket_missing:
+            locales:
+              en:
+                Oops! No bucket found.
+              es:
+                ¡Uy! No se ha encontrado el cubo.
+        - bucket_misconfigured:
+            locales:
+              en:
+                Hmm. The bucket is there, but it is misconfigured.
+              es:
+                Hmm. El cubo está allí, pero está mal configurado.
+      services:
+        - target_project.StorageV1
+      code: |-
```

Comment:
I'd like to also include a `type` or `code_type` field that indicates this step specifies its code as a Ruby method with particular arguments. GitWhisperer could infer it during bundling until we support more types but I think that would make it easier to extend this particular part without messing with the rest of the metadata (and revving the spec version) in the future.

Reply:
I'm open to that if others are strongly in favor of it, but my preference would be to leave it out for now. If/when it is useful, we can add a field for that in a backwards-compatible way (i.e. without changing the spec version) by just assuming `ruby` if not specified.